tailnet/coordinator.go

```diff
@@ -442,12 +508,66 @@ func (c *core) clientNodeUpdate(id, agent uuid.UUID, node *Node) error {
 
 	err := agentSocket.Enqueue([]*Node{node})
 	if err != nil {
-		return xerrors.Errorf("Enqueue node: %w", err)
+		return xerrors.Errorf("enqueue node: %w", err)
 	}
 	logger.Debug(context.Background(), "enqueued node to agent")
 	return nil
 }
 
+func (c *core) multiAgentSubscribe(enq Enqueueable, agentID uuid.UUID) (func(), error) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	c.initOrSetAgentConnectionSocketLocked(agentID, enq)
+
+	node, ok := c.nodes[enq.UniqueID()]
+	if ok {
+		// If we have the node, send it to the agent. If not, it will be sent
+		// async.
+		err := c.clientNodeUpdateLocked(enq.UniqueID(), agentID, node)
+		if err != nil {
+			return nil, xerrors.Errorf("send update to agent: %w", err)
+		}
+	} else {
+		c.logger.Debug(context.Background(), "multiagent node doesn't exist", slog.F("multiagent_id", enq.UniqueID()))
+	}
+
+	closer := func() {}
+
+	agentNode, ok := c.nodes[agentID]
+	if !ok {
+		// This is ok, once the agent connects the node will be sent over.
+		c.logger.Debug(context.Background(), "agent node doesn't exist", slog.F("agent_id", agentID))
+		return closer, nil
+	}
+
+	// Send the subscribed agent back to the multi agent.
+	err := enq.Enqueue([]*Node{agentNode})
+	return closer, err
+}
+
+func (c *core) multiAgentUpdate(id uuid.UUID, agents []uuid.UUID, node *Node) error {
```

Comment:
We still have this architecture where, even in the `core`, clients with a single agent and multiple agents are being handled independently.  From the perspective of the `core`, they should be identical.

In the code now, we've got this inconsistent handling, where for single-agent clients, the `core` is keeping track of the set (of size 1) of agents that the client is connecting to, and the core is governed by a mutex that makes it fairly easy to ensure that the right updates get sent out as things change.

However, for multi-agent clients, the core doesn't know all the agents the client is connecting to (technically it could compute it by inverting the map indexes on `agentToConnectionSockets`), and it instead depends on the `MultiAgent` to keep track and tell it.  But, this has resulted in a race condition bug that goes like this:

1. ServerTailnet calls `UpdateSelf()`, and we collect all the agents we should send an update to, and unlock the `MultiAgent` mutex.
2. In another goroutine, we call `SubscribeAgent()` with a new Agent.  This acquires the `MultiAgent` mutex, and then calls `core.MultiAgentSubscribe()`.  This locks the core, and sends the agent the stale node data (or, no node if step 1 above is the first time UpdateSelf() is being called).
3. The core unlocks, and the goroutine from step 1 calls `core.multiAgentUpdate()`.  However the agent from step 2 is not in the list, and it never gets the update.  Sadface.

I really think the core should be written to _only_ deal with clients that can have multiple agents, and the core should be responsible for all the state about who is connecting to who.  End-user clients that only have one agent should be modeled in the core as an initialization followed immediately by a subscribe.

You'll probably want to have a map that's the inverse of `agentToConnectionSockets`, where the first index is clientID and the second index is `agentID` --- this will make client node updates fast, since now we won't have the agentID on calls to `clientNodeUpdate()`